### PR TITLE
[Intel MKL] Adding support for MKL to docker CI infrastructure. 

### PR DESCRIPTION
**Pull request by claynerobison**
Creating a new PR because rebasing #18020 was a mess. See comments there.

### CODE REVIEW

File: tensorflow/tools/docker/parameterized_docker_build.sh [hunk 1]
```diff
@@ -299,6 +317,26 @@ else
...
+    then
+      echo "Modified Dockerfile to add MKL support."
+    else
+      die "FAILED to add MKL support to ${DOCKERFILE}."
+    fi
+  fi
+
+  # Modify the devel Dockerfile with bazel build options
+  if [[ ! -z ${TF_BAZEL_BUILD_OPTIONS} ]]; then
+    if sed -i -e "s/--cxxopt=\"-D_GLIBCXX_USE_CXX11_ABI=0\" /--cxxopt=\"-D_GLIBCXX_USE_CXX11_ABI=0\" ${TF_BAZEL_BUILD_OPTIONS} /g" "${DOCKERFILE}"
```
Comment (case540) [inline tensorflow/tools/docker/parameterized_docker_build.sh]: My comment still stands from before. I'd prefer if we could set the necessary Bazel options in a less hacky way.

File: tensorflow/tools/docker/parameterized_docker_build.sh [hunk 2]
```diff
@@ -299,6 +317,26 @@ else
   sed "s/^RUN git clone --branch=.* --depth=1/RUN git clone --branch=${TF_DOCKER_BUILD_DEVEL_BRANCH} --depth=1/" \
       "${ORIG_DOCKERFILE}" > "${DOCKERFILE}"
 
+  # Modify the devel Dockerfile to specify the MKL bazel build parameters
+  if [[ "${TF_DOCKER_BUILD_TYPE}" == "mkl" ]]; then
+    if sed -i -e 's/bazel build -c opt /bazel build -c opt --config=mkl /g' "${DOCKERFILE}"
```
Comment (case540) [inline tensorflow/tools/docker/parameterized_docker_build.sh]: My comment still stands from before. I'd prefer if we could set the necessary Bazel options in a less hacky way.

**Comment by claynerobison** [comment]
@case540 Don't bother. As you requested, I'm designing a new solution using Dockerfile ARG and --build-arg rather than sed.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @case540: It has been 16 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

File: tensorflow/tools/docker/parameterized_docker_build.sh (hunk 2 above)
Comment (congxu-ml) [reply tensorflow/tools/docker/parameterized_docker_build.sh]: I think using sed is an optimal solution for setting multiple Bazel options in Docker file, it makes code neat and elegant. This file also uses sed in other place. Any better suggestion?

Comment (congxu-ml) [reply tensorflow/tools/docker/parameterized_docker_build.sh]: @case540 The second option is using bazelrc file to pass Bazel options to Docker file, this method is used in
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/Dockerfile.cpu.mpi

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @case540: It has been 41 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by claynerobison** [comment]
Closing because https://github.com/tensorflow/tensorflow/pull/20164 was merged

**Pull request closed without merging**